app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+  has_many :images, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :commented_images, through: :comments, source: :image
+  validates :username,
+  :presence => true,
+  :uniqueness => {
+    :case_sensitive => false
+  }
+  validate :validate_username
+
+  def validate_username
+    if User.where(email: username).exists?
```

Comment:
Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)<br>Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)